returnn/flow.py

```diff
@@ -52,66 +83,112 @@ def make_precomputed_hybrid_tf_feature_flow(
+    rasr_config = rasr.RasrConfig()
+    rasr_config.input_map.info_0.param_name = "input"
+    rasr_config.input_map.info_0.tensor_name = f"extern_data/placeholders/{extern_data_name}/{extern_data_name}"
+    rasr_config.input_map.info_0.seq_length_tensor_name = (
         f"extern_data/placeholders/" f"{extern_data_name}/{extern_data_name}_dim0_size"
     )
 
-    tf_flow.config[tf_fwd].output_map.info_0.param_name = "log-posteriors"
-    tf_flow.config[tf_fwd].output_map.info_0.tensor_name = f"{output_layer_name}/output_batch_major"
+    rasr_config.output_map.info_0.param_name = "log-posteriors"
+    rasr_config.output_map.info_0.tensor_name = f"{output_layer_name}/output_batch_major"
 
-    tf_flow.config[tf_fwd].loader.type = "meta"
-    tf_flow.config[tf_fwd].loader.meta_graph_file = tf_graph
-    tf_flow.config[tf_fwd].loader.saved_model_file = tf_checkpoint
+    rasr_config.loader.type = "meta"
+    rasr_config.loader.meta_graph_file = tf_graph
+    rasr_config.loader.saved_model_file = tf_checkpoint
     if native_ops is not None:
         if isinstance(native_ops, list):
-            tf_flow.config[tf_fwd].loader.required_libraries = DelayedJoin(native_ops, ";")
+            rasr_config.loader.required_libraries = DelayedJoin(native_ops, ";")
         else:
-            tf_flow.config[tf_fwd].loader.required_libraries = native_ops
+            rasr_config.loader.required_libraries = native_ops
+    return make_precomputed_hybrid_feature_flow(
+        backend="tf",
+        rasr_config=rasr_config,
+        fwd_input_name=tf_fwd_input_name,
+    )
 
-    return tf_flow
 
+def make_precomputed_hybrid_onnx_feature_flow(
+    onnx_model: tk.Path,
+    io_map: Dict[str, str],
+    onnx_fwd_input_name: str = "onnx-fwd-input",
+    cpu: int = 1,
+) -> rasr.FlowNetwork:
+    """
+    Create the feature flow for a simple ONNX network that predicts frame-wise outputs,
+    see make_precomputed_hybrid_feature_flow.
 
-def add_tf_flow_to_base_flow(
+    With the config settings:
+
+        [flf-lattice-tool.network.recognizer.feature-extraction.onnx-fwd.io-map]
+        features      = data
+        features-size = data_len
+        output        = classes
+
+        [flf-lattice-tool.network.recognizer.feature-extraction.onnx-fwd.session]
+        file                 = <onnx_file>
+        inter-op-num-threads = 2
+        intra-op-num-threads = 2
+
+    :param onnx_model: usually the output of a OnnxExportJob
+    :param io_map: e.g. {"features": "data", "output": "classes"}
+    :param onnx_fwd_input_name: naming for the onnx network input, usually no need to be changed
+    :param cpu: number of CPUs to use
+    :return: onnx-forward node flow with output link and related config
+    """
+
+    rasr_config = rasr.RasrConfig()
+    for k, v in io_map.items():
+        rasr_config.io_map[k] = v
+
+    rasr_config.session.file = onnx_model
+    rasr_config.session.inter_op_num_threads = cpu
```

Comment:
This should better be in a non-hashed "post-config", as it has no influence on the result but is purely about speed optimization.

Reply:
Do we already have post configs for feature flows somewhere?

Reply:
I do not think so, but I checked the FlowNetwork class and it seems to support post configs.

Reply:
Sounds like we need a separate PR for that first?